### PR TITLE
Fix tests and chef18 fixes

### PR DESCRIPTION
**Pull request by logicsys**
# Description

Fix test cookbook by updating nginx dependency & creating an nginx_service resource

add unified_mode true & String type to connect & accept properties for connection resource

## Issues Resolved

None

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by Stromweld** [comment]
This needs the .github/workflows/ci.yml file updated to use the singular linting jobs and should look similar to this one https://github.com/sous-chefs/php/blob/main/.github/workflows/ci.yml

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +34,7 @@ jobs:
         uses: actionshub/markdownlint@main
 
   integration:
-    needs: [mdl, yamllint, delivery]
+    needs: [mdl, yamllint, lint-unit]
```
Comment (Stromweld) [inline .github/workflows/ci.yml:37]: lint-unit replaces the need for mdl, yamllint, and delivery jobs.

**Review by Stromweld — CHANGES_REQUESTED** [review]

File: resources/connection.rb [hunk 2]
```diff
@@ -17,11 +17,13 @@
...
 
+unified_mode true
+
 default_action :create
 
 property :service_name, String, name_property: true
-property :connect, required: true
-property :accept, required: true
+property :connect, String, required: true
+property :accept, String, required: true
```
Comment (Stromweld) [inline resources/connection.rb:26]: These actually should probably be both string or integer `[String, Integer]` then the test cookbooks don't need to be updated with quotes when it's just a port number.

**Review by Stromweld — CHANGES_REQUESTED** [review]

**Comment by logicsys** [comment]
Hmm, not sure why those last 5 tests are failing, working fine locally for me

**Comment by damacus** [comment]
Some of the yum and apt repositories have been getting hammered today. I had 1/2 of my suites fail because of that.

**Comment by logicsys** [comment]
think there was something up with the dokken ubuntu-1604 image and systemd support, have dropped that from the tests anyway, and added ubuntu-2004, ubuntu-2204 & rockylinux-8

File: test/cookbooks/test/metadata.rb [hunk 3]
```diff
@@ -2,4 +2,4 @@
 version '0.0.1'
 
 depends 'stunnel'
-depends 'nginx', '~> 10.0.2'
+depends 'nginx', '~> 12.1.1'
```
Comment (damacus) [inline test/cookbooks/test/metadata.rb:5]: Ah missed this. We can unpin this completely so we find the breakage sooner.

**Review by damacus — CHANGES_REQUESTED** [review]
Just one change on the Nginx dep and we're good to go!

**Comment by logicsys** [comment]
unpinned, that should be everything done finally ha

**Review by damacus — APPROVED** [review]

**Review by damacus — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [5.0.0](https://github.com/sous-chefs/stunnel/releases/tag/5.0.0)

**Pull request merged**